Common/src/linear_algebra/CSysSolve.cpp

```diff
@@ -945,15 +937,21 @@ unsigned long CSysSolve<ScalarType>::Solve(CSysMatrix<ScalarType> & Jacobian, co
 
     /*--- Build preconditioner for the transposed Jacobian ---*/
 
+    if (RequiresTranspose) Jacobian.TransposeInPlace();
+
```

Comment:
The transposition happens here.
Things that could go wrong with this approach:
- Solving the same system multiple times, without clearing the matrix or reverting the transposition in between. It does not happen atm.

I think it is worth the risk, since all preconditioners can be used now, and re-transposing the matrix if necessary is not very expensive (probably the same as applying the ILU once).

Reply:
Not that I used a preconditioner besides ILU for DA cases yet, but that is a nice thing 👍